src/Lecture3.hs

```diff
@@ -213,8 +238,15 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f acc (List1 x []) = f x acc
+  foldr f acc (List1 x (y:ys)) = foldr f (f y acc) (List1 x ys)
```

Comment:
You don't need to implement a manual recursion here, you can reuse the `Foldable` instance for lists 📂